schema/spacewalk/common/tables/suseRecurringAction.sql

```diff
@@ -0,0 +1,47 @@
+CREATE TABLE suseRecurringAction
+(
+  id                NUMERIC NOT NULL
+                    CONSTRAINT suse_recurring_action_id_pk PRIMARY KEY,
+  target_type       VARCHAR(32) NOT NULL,
+  name              VARCHAR(256) NOT NULL,
+  cron_expr         VARCHAR(32) NOT NULL,
+  minion_id         NUMERIC
+                    CONSTRAINT suse_rec_action_minion_fk
+                      REFERENCES suseMinionInfo(server_id)
+                      ON DELETE CASCADE,
+  group_id          NUMERIC
+                    CONSTRAINT suse_rec_action_group_fk
+                      REFERENCES rhnServerGroup(id)
+                      ON DELETE CASCADE,
+  org_id            NUMERIC
+                    CONSTRAINT suse_rec_action_org_fk
+                      REFERENCES web_customer(id)
+                      ON DELETE CASCADE,
+  creator_id        NUMERIC
+                    CONSTRAINT suse_rec_action_creator_fk
+                      REFERENCES web_contact(id)
+                      ON DELETE CASCADE,
+  active            CHAR(1) DEFAULT ('Y') NOT NULL,
+  test_mode         CHAR(1) DEFAULT ('Y') NOT NULL,
+  created           TIMESTAMP WITH TIME ZONE
+                      DEFAULT (current_timestamp)
+                      NOT NULL,
+  modified          TIMESTAMP WITH TIME ZONE
+                      DEFAULT (current_timestamp)
+                      NOT NULL
+);
+
+CREATE SEQUENCE suse_recurring_action_id_seq;
+
+CREATE INDEX suse_rec_action_type
+    ON suseRecurringAction(target_type);
+
+CREATE UNIQUE INDEX suse_rec_action_name_minion_uq
+    ON suseRecurringAction(name, minion_id);
+
+CREATE UNIQUE INDEX suse_rec_action_name_grp_uq
+    ON suseRecurringAction(name, group_id);
+
+CREATE UNIQUE INDEX suse_rec_action_name_org_uq
+    ON suseRecurringAction(name, org_id);
```

Comment:
AFAIK minion_id, group_id or org_id can be NULL. Postgresql unique indexes do not work well with NULL columns.
This requires special handling.

Example: schema/spacewalk/postgres/tables/rhnContentSourceSsl_index.sql